src/test/java/seedu/address/logic/commands/UnlabelCommandTest.java

```diff
@@ -0,0 +1,126 @@
+import static seedu.address.logic.commands.CommandTestUtil.VALID_LABEL;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import java.util.HashSet;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.label.Label;
+import seedu.address.model.tag.Tag;
+import seedu.address.model.tag.TagName;
+import seedu.address.testutil.TagBuilder;
+
+public class UnlabelCommandTest {
+    private ModelStubWithTag modelStub;
+
+    @BeforeEach
+    public void setUp() {
+        Tag newTag = new TagBuilder().build();
+        modelStub = new ModelStubWithTag(newTag);
+    }
+
+    @Test
+    public void equals() {
+        TagName oldTagName = new TagName("oldTag");
+        TagName newTagName = new TagName("newTag");
+
+        HashSet<Label> emptyLabels = new HashSet<>();
+        HashSet<Label> labels = new HashSet<>();
+        labels.add(new Label("testLabel"));
+
+        UnlabelCommand firstUnlabelCommand = new UnlabelCommand(newTagName, emptyLabels);
+        UnlabelCommand secondUnlabelCommand = new UnlabelCommand(oldTagName, labels);
+
+        // same object -> returns true
+        assertTrue(firstUnlabelCommand.equals(firstUnlabelCommand));
+
+        // same values -> returns true
+        UnlabelCommand firstUnlabelommandCopy = new UnlabelCommand(newTagName, emptyLabels);
+        assertTrue(firstUnlabelCommand.equals(firstUnlabelommandCopy));
+
+        // different types -> returns false
+        assertFalse(firstUnlabelCommand.equals(1));
+
+        // null -> returns false
+        assertFalse(firstUnlabelCommand.equals(null));
+
+        // different tagName -> returns false
+        assertFalse(firstUnlabelCommand.equals(secondUnlabelCommand));
+
+        // different labels -> returns false
+        assertFalse(firstUnlabelCommand.equals(secondUnlabelCommand));
+    }
+
+    @Test
+    public void execute_tagNotInModel_throwCommandException() {
+        TagName nonExistingTagName = new TagName("noExist");
+        HashSet<Label> labels = new HashSet<>();
+
+        UnlabelCommand unlabelCommand = new UnlabelCommand(nonExistingTagName, labels);
+
+        assertThrows(CommandException.class, String.format(UnlabelCommand.MESSAGE_TAG_NOT_FOUND,
+                nonExistingTagName.tagName), () -> unlabelCommand.execute(modelStub));
+    }
+
+    @Test
+    public void execute_emptyLabels_success() {
+        Tag newTag = new TagBuilder().build();
+
+        ModelStubWithTag expectedModelStub = new ModelStubWithTag(newTag);
+
+        HashSet<Label> labels = new HashSet<>();
+
+        UnlabelCommand unlabelCommand = new UnlabelCommand(newTag.getTagName(), labels);
+
+        assertCommandSuccess(unlabelCommand, modelStub,
+                String.format(unlabelCommand.MESSAGE_SUCCESS, newTag), expectedModelStub);
+    }
```

Comment:
Why is empty labels executed successfully? I thought empty labels will not be accepted by the parser.

Reply:
It is the behavior of the UnlabelCommand. It is not possible normally because the parser will throw an error, but if you can provide UnlabelCommand with an empty set, it will run successfully, with nothing changes. I saw a similar test case in the LabelCommand, so that's why I put it in the UnlabelCommand test too. I can delete it if you think it is unnecessary.

Reply:
Ooo, ok, I think it's fine to keep it